### PR TITLE
allow overriding of model_config parameters from the YML

### PR DESCRIPTION
**Pull request by winglian**
This should allow users to update values in the model config without having to edit the JSON.

for example for the attention_dropout for #672 
```
model_config:
    attention_dropout: 0.01
```

### CODE REVIEW

File: src/axolotl/utils/models.py [hunk 1]
```diff
@@ -358,13 +354,7 @@ def load_model(
             "Exception raised attempting to load model, retrying with AutoModelForCausalLM"
```
Comment (NanoCode012) [inline src/axolotl/utils/models.py]: Should remove this LOG.error then if you're removing the this AutoModel block.

**Review by NanoCode012 — COMMENTED** [review]
Nice clean PR! I think you'd want to add this to the docs as well. 

Should we deprecate the current rope configs and move them here also?

File: src/axolotl/utils/models.py [hunk 2]
```diff
@@ -234,20 +232,17 @@ def load_model(
         ):
             model_kwargs["use_flash_attention_2"] = True
 
+    if cfg.model_config:
+        for key, val in cfg.model_config.items():
+            setattr(model_config, key, val)
```
Comment (NanoCode012) [inline src/axolotl/utils/models.py]: I forgot to ask if this deals with nested items such as in rope_scaling case?

**Review by NanoCode012 — COMMENTED** [review]
One more point to update would be the config_validate for backward compatible..

File: src/axolotl/utils/models.py (hunk 2 above)
Comment (winglian) [reply src/axolotl/utils/models.py]: this would simply update the entire value of `rope_scaling`, so it's not recursive

**Comment by winglian** [comment]
> Should we deprecate the current rope configs and move them here also?

which rope configs are you referring to?

**Comment by NanoCode012** [comment]
> > Should we deprecate the current rope configs and move them here also?
> 
> which rope configs are you referring to?

Ah, I meant `rope_scaling`. Since we now expect them in the model_overide, we should add it in our config validation check, so that it doesn't break old configs and warns user of deprecation.

**Pull request merged**